Utilities/MCStepLogger/include/MCStepLogger/ROOTManager.h

```diff
@@ -0,0 +1,163 @@
+/* Wrapping some functionality of handling ROOT files
+ * Hides details of tree and directory access
+ */
+#ifndef ROOTMANAGER_H_
+#define ROOTMANAGER_H_
+
+#include <type_traits> // for std::is_pointer
+#include <string>
+#include <unordered_map>
+
+#include "TFile.h"
+#include "TTree.h"
+#include "TBranch.h"
+
+namespace o2
+{
+namespace mcstepanalysis
+{
+
+enum class ETFileMode : int { kREAD = 0,
+                              kUPDATE = 1,
+                              kRECREATE = 2 };
+
+class ROOTManager
```

Comment:
I find the name `ROOTManager` too generic. Could you come up with a more concret name?
Who is using this class? How many such objects are created?

(Maybe ROOTIOUtilities would be more appropriate)

Reply:
So far this class is used by the AnalysisManager to read the ROOT files produced by the MCStepLogger as well as by the AnalysisFile class to read/write the files after the analysis is done.